Graduate ReadWriteOncePod to beta, updated e2e test

chrishenzie opened the pull request:
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Graduates the ReadWriteOncePod feature from alpha to beta which now includes scheduler preemption handling for pods with this access mode. See https://github.com/kubernetes/kubernetes/pull/114051 for context. Updates the ReadWriteOncePod e2e test to handle the new behavior. Also updates the CSI hostpath driver to v1.11.0 which performs strict enforcement of the `SINGLE_NODE_SINGLE_WRITER` CSI access mode. See for more context:
https://github.com/kubernetes-csi/csi-driver-host-path/pull/381

#### Which issue(s) this PR fixes:
Fixes #114493

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
Graduate the `ReadWriteOncePod` feature gate to beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/2485
```

/sig storage
/sig scheduling
/milestone v1.27
/cc @msau42
/cc @jsafrane
/cc @alculquicondor
/hold

chrishenzie committed:
test: e2e test for ReadWriteOncePod preemption

chrishenzie committed:
feat: bump ReadWriteOncePod feature to beta

sftim commented:
Changelog text suggestion:
```
Graduate the `ReadWriteOncePod` feature gate to beta
```

chrishenzie committed:
chore: Update hostpath driver to v1.11.0

This version enforces the new SINGLE_NODE_SINGLE_WRITER CSI access mode
in NodePublishVolume.

See for more details:
https://github.com/kubernetes-csi/csi-driver-host-path/pull/381

chrishenzie commented:
/hold cancel
/assign @msau42

msau42 commented on test/e2e/storage/testsuites/readwriteoncepod.go:
```diff
@@ -45,9 +47,10 @@ var _ storageframework.TestSuite = &readWriteOncePodTestSuite{}
+	cs            clientset.Interface
+	volume        *storageframework.VolumeResource
+	pods          []*v1.Pod
+	priorityClass *schedulingv1.PriorityClass
```
Remove the "[Feature]" tag on L63 now that the feature is on by default.

msau42 commented:
/retest

msau42 commented:
/lgtm
/approve
